state-processor/src/main/java/com/ververica/flink/training/exercises/ReadRidesAndFaresSnapshot.java

```diff
@@ -0,0 +1,66 @@
+package com.ververica.flink.training.exercises;
+
+import org.apache.flink.api.common.state.ValueState;
+import org.apache.flink.api.common.state.ValueStateDescriptor;
+import org.apache.flink.api.java.DataSet;
+import org.apache.flink.api.java.ExecutionEnvironment;
+import org.apache.flink.api.java.tuple.Tuple2;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.runtime.state.memory.MemoryStateBackend;
+import org.apache.flink.state.api.ExistingSavepoint;
+import org.apache.flink.state.api.Savepoint;
+import org.apache.flink.state.api.functions.KeyedStateReaderFunction;
+import org.apache.flink.training.exercises.common.datatypes.TaxiFare;
+import org.apache.flink.training.exercises.common.datatypes.TaxiRide;
+import org.apache.flink.util.Collector;
+
+/**
+ * Java implementation for an example using the State Processor API to read and display
+ * the contents of a retained checkpoint or savepoint from RidesAndFaresSolution.
+ */
+public class ReadRidesAndFaresSnapshot {
+
+	/***************************************************************************************
+	 Update this path to point to a checkpoint or savepoint from RidesAndFaresSolution
+	 ***************************************************************************************/
+	public static final String PATH_TO_SNAPSHOT = "file:///tmp/checkpoints/58d523c8ae0ff19aba1053c4209d09bf/chk-1";
+
+	/**
+	 * Main method.
+	 *
+	 * @throws Exception which occurs during job execution.
+	 */
+	public static void main(String[] args) throws Exception {
+		ExecutionEnvironment bEnv = ExecutionEnvironment.getExecutionEnvironment();
```

Comment:
The troubleshooting exercises make use of `com.ververica.flink.training.common.EnvironmentUtils#createConfiguredEnvironment` here, which takes care of starting a local instance with a web UI, for example. Should we move this to the common project and use it here as well?

Reply:
In the developer training I prefer to keep things as vanilla as possible, so no, I'd rather not use that here unless you see some particular problem it would solve.